File: haskell/fibonacci.hs
--- Fibonacci Sequences in Haskell
---- https://code.golf/fibonacci#haskell
--- To be optimised for Code Golf
-- Pre-Golf:

main = putStrLn (printfmt (reverse (loopFibonacci 30)))

printfmt :: [Int] -> [Char]
printfmt [] = ""
printfmt (n:ns) = show n ++ "\n" ++ printfmt ns

loopFibonacci :: Int -> [Int]
loopFibonacci (-1) = []
loopFibonacci n = (fibonacci n) : loopFibonacci (n - 1)

fibonacci 0 = 0
fibonacci 1 = 1
fibonacci n = (fibonacci (n-1)) + (fibonacci (n-2))

-- Post-Golf:

--BEGIN--
main=putStrLn$p$reverse$l 30
p[]=""
p(n:m)=show n++'\n':p m
l(-1)=[]
l n=(f n):l(n-1)
f 0=0
f 1=1
f n=f(n-1)+f(n-2)
--END--

File: rust/src/fibonacci.rs
/*** Fibonacci Sequences in Rust
**** https://code.golf/fibonacci#rust
** Copyright (c) Kira K. 2022
*** To be optimised for Code Golf
*/

pub fn readable_main() {
    fn fib(n:u8) -> u32 {
        if n == 0 {return 0;}
        if n == 1 {return 1;}
        return fib(n-1) + fib(n-2);
    }
    for n in 0..31 {
        println!("{}", fib(n));
    }
}

pub/*BEGIN*/fn main(){fn f(n:u8)->u32{if n==0{return 0;}if n==1{return 1;}return f(n-1)+f(n-2);}for n in 0..31{println!("{}",f(n))}}/*END*/